includes/constants/class-base-constant.php

```diff
@@ -103,4 +103,14 @@ public static function __callStatic( $name, $arguments ) {
 	public function __toString() {
 		return constant( \get_class( $this ) . '::' . $this->get_value() );
 	}
+
+	/**
+	 * Specify the value which should be serialized to JSON.
+	 *
+	 * @return string
+	 */
+	#[\ReturnTypeWillChange]
```

Comment:
This `#[\ReturnTypeWillChange]` was necessary to prevent warnings like this: 

```
PHP Deprecated:  Return type of WCPay\Constants\Base_Constant::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /public/wp-content/plugins/woocommerce-payments/includes/constants/class-base-constant.php on line 112
```

I first attempted to add the return type to the signature but that caused other issues with the return type `mixed` not a valid type and I didn't want to restrict it to strings in case we intend on having int or other const types in the future. 

In any case there's already precedence for #[\ReturnTypeWillChange] being used in WCPay and widely used in WC core.